src/common/iconv.hh

```diff
@@ -40,18 +38,17 @@ public:
 
   QString convert( void const *& inBuf, size_t & inBytesLeft );
 
+  QString toUnicode( QUtf8StringView s );
+
```

Comment:
static method?

Reply:
I don't think this is an issue.

Each `Iconv` objects has its own `state` which, an integer descriptor pointing to internally managed things on libiconv's side. If there is thread issue, the convert in the above line also suffers as they are the same.

Reply:
when two threads invoke this toUnicode method ,  the state will be affected.

Reply:
Yes, but the one iconv belongs to SlobFile.

Every SLobFiles have its own instance of iconv.

If two threads may process the same file, something more wrong would already happen.

The original code's QTextCodec is also having internal states, so I think the correctness is unchanged.

Reply:
fulltext search and normal search should use the same slob dictionary instance

Reply:
ok